File: manual/data/java/tests/test_autopep8_parse_args.java
package com.example;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

import org.junit.jupiter.api.Test;

public class test_autopep8_parse_args {

    @Test
    public void test_parse_args_basic() {
        String input_str = "-n Alice -a 30";
        function_autopep8_parse_args parser = new function_autopep8_parse_args();
        String result = parser.parse_arguments(input_str);
        assertEquals("Name: Alice, Age: 30, City: Not provided", result);
    }

    @Test
    public void test_parse_args_with_city() {
        String input_str = "-n Bob -a 25 -c New York";
        function_autopep8_parse_args parser = new function_autopep8_parse_args();
        String result = parser.parse_arguments(input_str);
        // 因为 "New York" 被拆分成 "New" 和 "York"，因此返回错误信息 "Error: 2"
        assertEquals("Error: 2", result);
    }

    @Test
    public void test_parse_args_without_city() {
        String input_str = "-n Charlie -a 40";
        function_autopep8_parse_args parser = new function_autopep8_parse_args();
        String result = parser.parse_arguments(input_str);
        assertEquals("Name: Charlie, Age: 40, City: Not provided", result);
    }

    @Test
    public void test_parse_args_missing_age() {
        String input_str = "-n Dave";
        function_autopep8_parse_args parser = new function_autopep8_parse_args();
        String result = parser.parse_arguments(input_str);
        assertTrue(result.contains("Error:"));
    }

    @Test
    public void test_parse_args_missing_name() {
        String input_str = "-a 50";
        function_autopep8_parse_args parser = new function_autopep8_parse_args();
        String result = parser.parse_arguments(input_str);
        assertTrue(result.contains("Error:"));
    }
}

File: manual/data/cpp/tests/test_autopep8_parse_args.cpp
#include <gtest/gtest.h>
#include "../src/function_autopep8_parse_args.cpp"

class TestArgumentParser : public ::testing::Test {
protected:
    ArgumentParser parser;
};

TEST_F(TestArgumentParser, ParseArgsBasic) {
    std::string input_str = "-n Alice -a 30";
    std::string result = parser.parse_arguments(input_str);
    EXPECT_EQ(result, "Name: Alice, Age: 30, City: Not provided");
}

TEST_F(TestArgumentParser, ParseArgsWithCity) {
    std::string input_str = "-n Bob -a 25 -c New York";
    std::string result = parser.parse_arguments(input_str);
    EXPECT_EQ(result, "Error: 2");
}

TEST_F(TestArgumentParser, ParseArgsWithoutCity) {
    std::string input_str = "-n Charlie -a 40";
    std::string result = parser.parse_arguments(input_str);
    EXPECT_EQ(result, "Name: Charlie, Age: 40, City: Not provided");
}

TEST_F(TestArgumentParser, ParseArgsMissingAge) {
    std::string input_str = "-n Dave";
    std::string result = parser.parse_arguments(input_str);
    EXPECT_TRUE(result.find("Error:") != std::string::npos);
}

TEST_F(TestArgumentParser, ParseArgsMissingName) {
    std::string input_str = "-a 50";
    std::string result = parser.parse_arguments(input_str);
    EXPECT_TRUE(result.find("Error:") != std::string::npos);
}

int main(int argc, char **argv) {
    testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}